lib/hashes.md

```diff
@@ -1,44 +1,33 @@
 
 # How do you retrieve a particular value from a Hash?
+To retrieve a value, you just need the name of the hash and the name of the key whose value you are trying to find. Example foods[:pizza] would return => “best”
+(Shannon question: this doesn’t work the opposite way. How would I search for “best” on a hash of foods?)
 
-Replace me with your response...
-
-# How do you add information to a Hash?
+(# How do you add information to a Hash? # How would you change the value of a particular element in a Hash? # How do you delete an element from a Hash? )
 
-Replace me with your response...
+You can easily add new things to a hash by just telling ruby what the new key equals. To the above list, I can add foods[:nachos] = “second best” and it will add it to the pile. Since every key in the pile has a specific name, if you do that to a key that is ALREADY in the pile, it will change the value of a pre-existing hash: foods[:pizza] = “best EVER” (or changing the price on a menu, or increasing the grade of a student, etc) If you want to REMOVE items from a hash, you use .delete (if, say, you develop a tragic adult onset tomato allergy, and have to perform foods.delete(":pizza”), it will remove it from the list)
 
 # How would you perform an operation on every element inside a Hash?
 
-Replace me with your response...
-
-# How would you change the value of a particular element in a Hash?
-
-Replace me with your response...
-
-# How do you delete an element from a Hash?
-
-Replace me with your response...
+You can use .each/do to perform an operation on every elements. For the food list example:
+ foods.each do | item, value |
+  puts "#{item} = #{value}" + " but better with cheese"
+end
+And then you basically have a recipe book.
 
 # What happens if you try to retrieve an element from a Hash that does not exist in the Hash?
 
-Replace me with your response...
+If you try to retrieve an element that isn’t there, you get the answer of “nil” (meaning “there’s nothing by that name in the pile”) If you want there to be a specific response to nonexistent elements, you can assign a .default. For my previous example, you could assign foods.default = “we don’t care about that food” so when someone asks foods[:eggplant] they will get that as a response instead of “nil"
 
 # How do you determine how many elements are in a Hash?
-
-Replace me with your response...
\ No newline at end of file
+I’m not sure I have the most elegant answer to this question, but right now the way i would do it is by using .keys. Continuing this example, I could do foods.keys and it would return “pizza, nachos, french fries, toast” and then I could just count those.
```

Comment:
Oh, by count, I literally meant the human way. 
foods = { ":pizza" => "best", ":nachos" => "second best", ":toast" => "fine" }
foods.keys
and then count the returned keys, probably by moving my finger across my screen. Not only is this not an elegant solution, but as your array gets bigger, it would become less and less possible. foods.length is a much better solution. "Let the computer do it" is a motto I need to remember/listen to.